File: Application/dotNET/DataLayer/Models/AnomalyFlag.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataLayer.Models
{
    internal class AnomalyFlag
    {
       public long Anomaly_Flag_Id { get; set; }
       public long? Tanker_Id { get; set; }
       public long? Position_Id { get; set; }
       public long? Staging_Id { get; set; }
       public long Anomaly_Type_Id { get; set; }
       public string Source { get; set; } = "system";
       public decimal? Confidence { get; set; }
       public string? Notes { get; set; }
       public DateTime? Created_At { get; set; }
    }
}

File: Application/React_Frontend/src/types/AnomalyFlag.ts
export interface AnomalyFlag {
  Anomaly_Flag_Id: number;

  Tanker_Id: number | null;
  Position_Id: number | null;
  Staging_Id: number | null;

  Anomaly_Type_Id: number;

  Source: string;

  Confidence: number | null;

  Notes: string | null;

  Created_At: string | null;
}